scripts/upgrade-legacy-rules.py

```diff
@@ -0,0 +1,263 @@
+    """
+    Takes in an old rule and returns its equivalent in the new rule format.
+    """
+    features = content["rule"]["features"]
+
+    for key, value in features[0].items():
+        pass
+    stat, dyn = rec_features_list([{key: value}])
+
+    meta = update_meta(content["rule"]["meta"], has_dyn=dyn)
+    if dyn:
+        features = dyn
+    else:
+        features = stat
+
+    content["rule"] = {"meta": meta, "features": {"~": features}}
+    upgraded_rule = yaml.dump(content, Dumper=NoAliasDumper, sort_keys=False, width=float("inf")).split("\n")
+    upgraded_rule = "\n".join(list(filter(lambda line: "~" not in line, upgraded_rule)))
+    if Rule.from_yaml(upgraded_rule):
+        return upgraded_rule
+
+
+def main(argv: Optional[List[str]] = None) -> int:
+    desc = (
+        "Upgrade legacy-format rulesets into the new rules format which supports static and dynamic analysis flavors."
+    )
+    parser = argparse.ArgumentParser(description=desc, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument(
+        "--old-rules-path", default=Path(__file__).parents[1].joinpath("rules"), help="path to the legacy ruleset"
+    )
+    parser.add_argument(
+        "--save-path",
+        default=Path(__file__).parents[1].joinpath("upgraded-rules"),
+        help="where to save the upgraded rules",
+    )
+    args = parser.parse_args(args=argv)
+
+    # check args
+    old_rules_path = Path(args.old_rules_path)
+    new_rules_save_path = Path(args.save_path)
+
+    if old_rules_path == new_rules_save_path:
+        print(
+            textwrap.dedent(
+                """
+                WARNING: you've specified the same directory for the old-rules' path and the new rules' save path.
+                This will cause this script to overwrite your old rules with the new upgraded ones.
+                Are you sure you want proceed with overwritting the old rules [O]verwrite/[E]xit:
+                """
+            )
+        )
+        response = ""
+        while response not in ("o", "e"):
+            response = input().lower()
+            if response == "o":
+                print("Old rules' folder will be overwritten.")
+            elif response == "e":
+                print("The ruleset will not been upgraded.")
+                return 0
+            else:
+                print("Please provide a valid answer [O]verwrite/[E]xit: ")
+
+    # Get rules
+    rule_file_paths: List[Path] = collect_rule_file_paths([old_rules_path])
+    rule_contents = [rule_path.read_bytes() for rule_path in rule_file_paths]
+    for path, content in zip(rule_file_paths, rule_contents):
+        """
+        This loop goes through the list of rules and does the following:
+        1. Get the current rule's content.
+        2. Get its dynamic-format equivalent.
+        3. Compute its save path and save it there.
+        """
+        content = yaml.load(content.decode("utf-8"), Loader=yaml.Loader)
```

Comment:
see capafmt script, may be helpful to also do the below to keep comments and formatting
https://github.com/mandiant/capa/blob/7e78133925656ecc69b960cbb60e39c339e4ce81/scripts/capafmt.py#L60-L61

Reply:
ruamel is an easy solution for keeping comments and formatting, but I couldn't get around it dumping aliases and pointers instead of actual data. Like this:
![image](https://github.com/mandiant/capa/assets/16624109/75a04eb9-4b87-4a2c-8fb0-fb5590818da2)

Reply:
huh, that's strange, how are you dumping that?

Reply:
I've tried in several ways before but couldn't get rid of the aliases. here's what produced the output above (I got it from the `to_yaml()` method):
![image](https://github.com/mandiant/capa/assets/16624109/1b30f0e0-ba89-468b-bf96-16892577c965)

here's how I loaded:
![image](https://github.com/mandiant/capa/assets/16624109/dc8dda98-5bcd-44ce-ac6e-4ea4a3146ae4)

Reply:
Hm, too bad. I think getting that to work could save us a lot of trouble handling special cases...